packages/flutter/lib/src/material/range_slider.dart

```diff
@@ -1330,32 +1437,37 @@ class _RenderRangeSlider extends RenderBox with RelayoutWhenSystemFontsChangeMix
           break;
       }
 
-      _sliderTheme.rangeValueIndicatorShape.paint(
-        context,
-        topThumbCenter,
-        activationAnimation: _valueIndicatorAnimation,
-        enableAnimation: _enableAnimation,
-        isDiscrete: isDiscrete,
-        isOnTop: thumbDelta < innerOverflow,
-        labelPainter: topLabelPainter,
-        parentBox: this,
-        sliderTheme: _sliderTheme,
-        textDirection: _textDirection,
-        thumb: topThumb,
-        value: topValue,
-      );
+      _state.paintTopValueIndicator = (PaintingContext context, Offset offset) {
```

Comment:
This no longer needs to happen at this point of the paint method. Maybe move to the end for clarity?